### PR TITLE
CTAP2

### PR DESCRIPTION
**Pull request by daeMOn63**

### CODE REVIEW

File: ctap2token/token.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+}
+
+type PinProtocolVersion uint
+
+const (
+	PinProtoV1 PinProtocolVersion = 1
+)
+
+// MakeCredentialResponse...
+// CTAP 2.1 defines Fmt=0x1 and AuthData=0x2 while CTAP 2.0 defines AuthData=0x1 and Fmt=0x2 for some reasons
```
Comment (titanous) [inline ctap2token/token.go]: This appears to be a typo in the 2.0 spec. The `authenticatorMakeCredential_Response` table in both list fmt=0x01 and authData=0x02.

File: ctap2token/example/main.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+			panic(err)
+		}
+
+		userID := make([]byte, 32)
+		if _, err := rand.Read(userID); err != nil {
+			panic(err)
+		}
+
+		mac := hmac.New(sha256.New, pinHashDec)
+		_, err = mac.Write(clientDataHash)
```
Comment (titanous) [inline ctap2token/example/main.go]: The host side of the PIN protocol should be implemented in a separate package instead of inline in the example.

**Review by titanous — APPROVED** [review]

File: ctap2token/example/main.go (hunk 2 above)
Comment (daeMOn63) [reply ctap2token/example/main.go]: :+1: Moved all of it under a `pin` package

**Comment by daeMOn63** [comment]
included in webauthn branch, this version is now outdated.

**Pull request closed without merging**